frontend/src/data/media-service.ts

```diff
@@ -28,6 +29,45 @@ class MediaService<T extends Media> {
     this.mediaType = mediaType
   }
 
+  /**
+   * Processes AxiosResponse from a search query to
+   * construct SEARCH_RESPONSE_TIME analytics event.
+   * @param response - Axios response
+   * @param requestDatetime - datetime before request was sent
+   */
+  recordSearchTime(response: AxiosResponse, requestDatetime: Date) {
+    const { sendCustomEvent } = useAnalytics()
+    const REQUIRED_HEADERS = ["date", "cf-cache-status", "cf-ray"]
+
+    const responseHeaders = response.headers
+    if (!REQUIRED_HEADERS.every((header) => header in responseHeaders)) {
+      return
+    }
+
+    const responseDatetime = new Date(responseHeaders["date"])
+    if (responseDatetime < requestDatetime) {
+      // response returned was from the cache
+      return
+    }
+
+    const cfRayIATA = responseHeaders["cf-ray"].split("-")[1]
+    if (cfRayIATA === undefined) {
+      return
+    }
+
+    const elapsedSeconds = Math.floor(
+      (responseDatetime.getTime() - requestDatetime.getTime()) / 1000
+    )
+    const url = new URL(response.request?.responseURL)
+
+    sendCustomEvent("SEARCH_RESPONSE_TIME", {
+      cfCacheStatus: responseHeaders["cf-cache-status"],
+      cfRayIATA: cfRayIATA,
+      elapsedTime: elapsedSeconds,
+      queryString: url.search,
```

Comment:
All sounds good - I think having separate events makes sense to increase how usable the results are, so will get working on those changes. Just wanted to clarify on the second point - I was under the impression the Date header would help to clean out any local caching, rather than any Cloudflare caching. Do you mean that the local caching is something we'd also potentially like to track?

Reply:
Oh right! I completely forgot about that, you're right. The Cloudflare responses, even cached ones, will have dates not in the past.

I wrote the issue and the details of how it would work, and forgot this, so it would be good to clarify the comment to say "local cache" rather than just "cache". Our Cloudflare configuration caches some search responses, so the confusion will be easy to make for maintainers familiar with Openverse's infrastructure.

Reply:
> it might be better to use separate event names per media type

A lot of current analytics events use a single event with a subfield for the media type (seen by searching usages of type `SearchType` in the file `frontend/src/types/analytics.ts`) so I think it is acceptable to do so here (or we might want to go back and update the existing ones?)